src/runtime_src/core/tools/xbreplay/src/main.cpp

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#include "seq_reconstructor.hpp"
+#include "utils/cmd_args.hpp"
+#include "utils/logger.hpp"
+
+#include <iostream>
+#include <memory>
+namespace {
+using namespace xrt::tools::xbreplay;
+
+class successfull_termination : public std::exception
+{
+  public:
+  const char* what() const noexcept override
+  {
+    return "Program terminated successfully";
+  }
+};
+
+/*
+ * This function is used to parse the command line arguments
+ */
+static std::pair<std::string, std::string>
+parse_command_line_arguments(std::vector<std::string>& cmd_params)
+{
+  std::string trace_file;
+  std::string mem_file;
+  std::vector<std::string>& args = cmd_params;
+  cmd_args_opt opt;
+  std::vector<cmd_args_opt> options =
+  {
+    {'h', false, "", "To provide usage information"},
+    {'t', true, "", "To provide path to the trace file as input"},
+    {'d', true, "", "To provide path to the memory dump file"},
+    {'l', true, "", "To set the log level (DEBUG=0, INFO=1, WARN=2, ERROR=3)"}
+  };
+
+  cmd_args cargs(std::move(options));
```

Comment:
Why `options` and then immediately moved to `cargs`?

Reply:
cmd_args module should be able to accept the command line options as input. So creating options outside